tests/Export/ProductServiceTest.php

```diff
@@ -421,4 +423,224 @@ public function testProductIsNotFoundIfMainVariantIsNotAvailableForSalesChannel(
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+                [
+                    'option' => [
+                        'id' => $secondOptionId,
+                        'name' => 'Orange',
+                        'group' => [
+                            'id' => $optionGroupId,
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+                [
+                    'option' => [
+                        'id' => $thirdOptionId,
+                        'name' => 'Green',
+                        'group' => [
+                            'id' => $optionGroupId,
+                            'name' => 'Color',
+                        ],
+                    ],
+                ],
+            ],
+            'configuratorGroupConfig' => [
+                [
+                    'id' => $optionGroupId,
+                    // This tells Shopware to consider the mainVariationId if set.
+                    'expressionForListings' => $config === 'parent',
+                    'representation' => 'box'
+                ]
+            ],
+        ]);
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedFirstVariantId,
+            'parentId' => $parentId,
+            'productNumber' => 'FINDOLOGIC001.1',
+            'name' => 'FINDOLOGIC VARIANT 1',
+            'price' => [
+                [
+                    'currencyId' => Defaults::CURRENCY,
+                    'gross' => $cheapestPrice,
+                    'net' => $cheapestPrice / 2,
+                    'linked' => false
+                ]
+            ],
+            'options' => [
+                ['id' => $firstOptionId]
+            ],
+        ]));
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedSecondVariantId,
+            'parentId' => $parentId,
+            'productNumber' => 'FINDOLOGIC001.2',
+            'name' => 'FINDOLOGIC VARIANT 2',
+            'price' => [['currencyId' => Defaults::CURRENCY, 'gross' => 4, 'net' => 2, 'linked' => false]],
+            'options' => [
+                ['id' => $secondOptionId]
+            ],
+        ]));
+
+        $this->createVisibleTestProduct($this->getBasicVariantData([
+            'id' => $expectedThirdVariantId,
+            'parentId' => $parentId,
+            'productNumber' => 'FINDOLOGIC001.3',
+            'name' => 'FINDOLOGIC VARIANT 3',
+            'price' => [['currencyId' => Defaults::CURRENCY, 'gross' => 4, 'net' => 2, 'linked' => false]],
+            'options' => [
+                ['id' => $thirdOptionId]
+            ],
+        ]));
+
+        // Explicitly assign the main variant ID to the product, and assert that the assigned main variant
+        // is the exported variant.
+        if ($config === 'parent') {
```

Comment:
It is quite hard to read and understand everything this test is checking. Please try to move the test setup into helper methods and things that happen in the `if` blocks into separate tests. In this it probably also makes sense to split the data provider into three separate ones. It is always better to have many small tests instead of one big one. 

In an ideal world a test should also have only one or two assertions (but this is more like a rule of thumb).